kmath-core/src/commonMain/kotlin/space/kscience/kmath/operations/Algebra.kt

```diff
@@ -131,7 +123,7 @@ public interface SpaceOperations<T> : Algebra<T> {
      * @receiver this value.
      * @return the additive inverse of this value.
      */
-    public operator fun T.unaryMinus(): T = multiply(this, -1.0)
+    public operator fun T.unaryMinus(): T
```

Comment:
A separate negate(T) method can be added to improve unaryOperationFunction performance by returning function reference ::negate (unfortunately, no method references to operator functions can be created). In the future, method references will be a bit faster than lambdas because of https://youtrack.jetbrains.com/issue/KT-26060

Reply:
It should be done in a separate PR